modules/sql/servers/main.bicep

```diff
@@ -176,7 +176,7 @@ module server_databases 'databases/main.bicep' = [for (database, index) in datab
     diagnosticEventHubAuthorizationRuleId: contains(database, 'diagnosticEventHubAuthorizationRuleId') ? database.diagnosticEventHubAuthorizationRuleId : ''
     diagnosticEventHubName: contains(database, 'diagnosticEventHubName') ? database.diagnosticEventHubName : ''
     isLedgerOn: contains(database, 'isLedgerOn') ? database.isLedgerOn : false
-    location: contains(database, 'location') ? database.location : server.location
+    location: contains(database, 'location') ? database.location : location
```

Comment:
Hi @thofle thanks much for raising this!

If databases are not supported in a different location than their parent server, I agree to the suggestion you are providing in the PR description and that we should probably directly pass the parent location to its child.

If that is the case, we could go one step further and 
- remove the location parameter from the database child module, which defaults to the rg location and not its parent one
- when assigning the location to the resource database in the child database module, use server.location in [this line](https://github.com/Azure/ResourceModules/blob/main/modules/sql/servers/databases/main.bicep#L224), unless that would impact what-if in the same way
- regenerate readme files to comply with the changes

What's your take on that? I think the same principle may apply to other resources as well (e.g. NetApps) of course out of the scope of this PR

Reply:
Hi @eriqua 

The location parameter is required for the database deployment ([docs](https://learn.microsoft.com/en-us/azure/templates/microsoft.sql/2022-05-01-preview/servers/databases?pivots=deployment-language-arm-template)) so it can't be removed from the child module. And it cannot reference the location of the of the server in the child module as you suggested since it requires a value that can be calculated at the start of the deployment.

Maybe it would be better just to change `location: contains(database, 'location') ? database.location : location` to `location: location` in the server deployment?

I just did a test with elastic pool as well and it has the same issue. It does not show up in what-if results if server.location is used and it also needs to be in the same location as the server to deploy successfully.